WalletWasabi/Blockchain/Keys/WalletGenerator.cs

```diff
@@ -0,0 +1,69 @@
+using NBitcoin;
+using System;
+using System.IO;
+using System.Linq;
+using WalletWasabi.Helpers;
+using WalletWasabi.Models;
+
+namespace WalletWasabi.Blockchain.Keys
+{
+	public class WalletGenerator
+	{
+		private static readonly string[] ReservedFileNames = new string[]
+		{
+			"CON", "PRN", "AUX", "NUL",
+			"COM1", "COM2", "COM3", "COM4", "COM5", "COM6", "COM7", "COM8", "COM9",
+			"LPT1", "LPT2", "LPT3", "LPT4", "LPT5", "LPT6", "LPT7", "LPT8", "LPT9",
+		};
+
+		public WalletGenerator(string walletsDir, Network network)
+		{
+			WalletsDir = walletsDir;
+			Network = network;
+		}
+
+		public string WalletsDir { get; private set; }
+		public Network Network { get; private set; }
+		public bool TermsAccepted { get; set; }
+		public uint TipHeight { get; set; }
+
+		public (KeyManager, Mnemonic) GenerateWallet(string walletName, string password)
```

Comment:
This is used as a static factory method. Consider converting it to static?

Reply:
I don't agree. Many classes started like this one having only one method that looks like a static method (eg: TransactionProcessor) and then evolved. We should avoid static classes imo.